libs/core/algorithms/include/hpx/parallel/util/loop.hpp

```diff
@@ -125,6 +127,39 @@ namespace hpx::parallel::util {
         };
     }    // namespace detail
 
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail {
+
+        // Helper class to repeatedly call a function starting from a given
+        // iterator position till the predicate returns true.
+        template <typename Iterator, typename ExPolicy>
+        struct loop_pred
+        {
+            ///////////////////////////////////////////////////////////////////
+            template <typename Begin, typename End, typename Pred>
+            HPX_HOST_DEVICE HPX_FORCEINLINE static constexpr Begin call(
+                Begin it, End end, Pred&& pred)
+            {
+                if constexpr (hpx::is_unsequenced_execution_policy_v<
```

Comment:
Could we move this dispatch point to the CPO below, please?

Reply:
moved to tag_invoke